variable_set.go

```diff
@@ -63,43 +72,39 @@ const (
 	VariableSetVars       VariableSetIncludeOpt = "vars"
 )
 
+// VariableSetListOptions represents the options for listing variable sets.
 type VariableSetListOptions struct {
 	ListOptions
 	Include string `url:"include"`
 }
 
-func (o *VariableSetListOptions) valid() error {
-	return nil
-}
+// VariableSetCreateOptions represents the options for creating a new variable set within in a organization.
+type VariableSetCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,varsets"`
 
-// List all Variable Sets in the organization
-func (s *variableSets) List(ctx context.Context, organization string, options *VariableSetListOptions) (*VariableSetList, error) {
-	if !validStringID(&organization) {
-		return nil, ErrInvalidOrg
-	}
-	if options != nil {
-		if err := options.valid(); err != nil {
-			return nil, err
-		}
-	}
+	// The name of the variable set.
+	// Affects variable precedence when there are conflicts between Variable Sets
+	// https://www.terraform.io/cloud-docs/api-docs/variable-sets#apply-variable-set-to-workspaces
+	Name *string `jsonapi:"attr,name"`
 
-	u := fmt.Sprintf("organizations/%s/varsets", url.QueryEscape(organization))
-	req, err := s.client.newRequest("GET", u, options)
-	if err != nil {
-		return nil, err
-	}
+	// A description to provide context for the variable set.
+	Description *string `jsonapi:"attr,description,omitempty"`
 
-	vl := &VariableSetList{}
-	err = s.client.do(ctx, req, vl)
-	if err != nil {
-		return nil, err
-	}
+	// If true the variable set is considered in all runs in the organization.
+	Global *bool `jsonapi:"attr,global,omitempty"`
```

Comment:
This appears to be an optional value (pointer, omitempty) but we make sure the value isn't nil within VariableSetCreateOptions valid method

Reply:
Hey! I can't speak to this particular type. It appears in the diff because I re-ordered everything in variable_set.go at Sebastian's request, but I'm not sure I have the necessary context to understand the ask here or competently resolve it. It might make for a good GH issue to handle in a separate PR.